eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml

```diff
@@ -18,15 +18,29 @@ jobs:
   variables:
     displayNameFilter: $[ coalesce(variables.jobMatrixFilter, '.*') ]
   pool:
-    name: Azure Pipelines
-    vmImage: ubuntu-18.04
+    name: azsdk-pool-mms-ubuntu-1804-general
+    vmImage: MMSUbuntu18.04
   displayName: Generate Job Matrix
   steps:
+    - checkout: none
+
+    - pwsh: |
+        git clone --no-checkout --filter=tree:0 git://github.com/$(Build.Repository.Name).git .
+        git sparse-checkout init
+        if ("${{ parameters.AdditionalParameters.ServiceDirectory }}") {
+          git sparse-checkout set eng sdk/${{ parameters.AdditionalParameters.ServiceDirectory}}
+        } else {
+          git sparse-checkout set eng
```

Comment:
Will this ever be enough? It seems like we would at least need to checkout the directory where the matrix file is located.

Reply:
If there's no service directory, then I assume we're using the default matrix config which is located in `eng`.